processor/src/stack/mod.rs

```diff
@@ -161,13 +161,38 @@ impl Stack {
     /// stack will be at the last position in the word.
     ///
     /// Creating a word does not change the state of the stack.
-    pub fn get_word(&self, word_idx: usize) -> Word {
-        let offset = word_idx * WORD_SIZE;
+    pub fn get_word(&self, idx: usize) -> Word {
+        [self.get(idx + 3), self.get(idx + 2), self.get(idx + 1), self.get(idx)].into()
+    }
```

Comment:
Doc comments for this function need to be updated.

Also, changing semantics in this way may lead to difficult-to-detect bugs downstream. At the very least, we should clearly document this as a breaking change in the changelog. But I also wonder if a different name should be used instead.